plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -282,6 +282,10 @@ of them that are important to know to be able to use 'makeBuiltinMeaning' in cas
 complicated than a simple monomorphic or polymorphic function. But for now let's talk about a few
 more simple cases.
 
+Note that this only applies when each unconstrained type variable is only used as the entire
+type of one or more arguments, as is the case with @\b x y -> if b then x else y@. Otherwise,
```

Comment:
That's not true. It's fine to apply `EvaluationResult` or `Emitter` or `Opaque` or any combination of those to a type variable.